### PR TITLE
test: Nullable fields for SentrySpanContext

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Add a test to verify that the serialization of SentrySpanContext handles nil values.

#skip-changelog

## :bulb: Motivation and Context

https://github.com/getsentry/sentry-cocoa/pull/1111#discussion_r642999769

## :green_heart: How did you test it?
I added tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Tests/SentryTests/SentrySpanContextTests.swift [hunk 1]
```diff
@@ -62,6 +62,17 @@ class SentrySpanContextTests: XCTestCase {
         XCTAssertEqual(data["status"] as? String, "ok")
     }
     
+    func testSerialization_NotSettingProperties_PropertiesNotSerialized() {
+        let spanContext = SpanContext(operation: someOperation)
+        
+        let data = spanContext.serialize()
+        
+        XCTAssertNil(data["description"])
+        XCTAssertNil(data["sampled"])
```
Comment (bruno-garcia) [inline Tests/SentryTests/SentrySpanContextTests.swift:71]: Isn't this required value?

Comment (philipphofmann) [reply Tests/SentryTests/SentrySpanContextTests.swift:71]: No, it's also nullable in java, see https://github.com/getsentry/sentry-java/blob/ee459986fe336dc1e0a11fdb0bf472ec90cb2de4/sentry/src/main/java/io/sentry/SpanContext.java#L27.

**Review by bruno-garcia — APPROVED** [review]
